src/main/java/eod/warObject/character/abstraction/Character.java

```diff
@@ -146,15 +154,27 @@ public void die() {
         teardown();
     }
 
+    @Override
+    public int getHp() {
+        return hp;
+    }
+
     @Override
     public void teardown() {
         super.teardown();
         status.clear();
     }
 
     @Override
-    public int getHp() {
-        return hp;
+    public void onEventOccurred(GameObject sender, Event event) {
+        super.onEventOccurred(sender, event);
+        if(event instanceof RoundStartEvent) {
+            if(((RoundStartEvent) event).getStartedRound().getPlayer().isPlayerA() == player.isPlayerA() && hasStatus(POISON)) {
```

Comment:
Player has a `isActingPlayer` property, I think it will fit this use.
You can get access to it by creating a public getter for it.

Reply:
solved